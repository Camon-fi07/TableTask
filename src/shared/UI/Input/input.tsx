import clsx from 'clsx';

import styles from './input.module.css';
import { InputProps } from './input.types';

export const Input = ({
  leftIcon,
  rightIcon,
  isError,
  ...props
}: InputProps) => (
  <div
    data-testid='input-wrapper'
    className={`${styles.wrapper} ${isError ? styles.error : ''}`}
  >
    {!!leftIcon && <div className={styles.icon}>{leftIcon}</div>}
    <input
      {...props}
      className={clsx(
        styles.input,
        { [styles.inputWithoutIcons]: !leftIcon && !rightIcon },
        props.className
      )}
    />
    {!!rightIcon && <div className={styles.icon}>{rightIcon}</div>}
  </div>
);
