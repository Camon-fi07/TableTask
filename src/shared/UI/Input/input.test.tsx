import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { EyeIcon, SearchIcon } from 'lucide-react';

import { Input } from './input';

describe('Input', () => {
  it('should render', () => {
    const { getByTestId } = render(<Input data-testid='input' />);

    expect(getByTestId('input')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { getByTestId } = render(
      <Input data-testid='input' placeholder='placeholder' />
    );

    expect(getByTestId('input')).toMatchSnapshot();
  });

  it('should match snapshot with icons', () => {
    const { getByTestId } = render(
      <Input
        data-testid='input'
        placeholder='placeholder'
        leftIcon={<SearchIcon />}
        rightIcon={<EyeIcon />}
      />
    );

    expect(getByTestId('input')).toMatchSnapshot();
  });

  it('should change value', async () => {
    const { getByTestId } = render(<Input data-testid='input' />);

    const input = getByTestId('input');

    await userEvent.type(input, 'test');

    expect(input).toHaveValue('test');
  });
});
