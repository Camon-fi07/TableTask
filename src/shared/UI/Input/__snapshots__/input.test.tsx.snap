// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<input
  class="input inputWithoutIcons"
  data-testid="input"
  placeholder="placeholder"
/>
`;

exports[`Input should match snapshot with icons 1`] = `
<input
  class="input"
  data-testid="input"
  placeholder="placeholder"
/>
`;

exports[`Input should render 1`] = `
<input
  class="input inputWithoutIcons"
  data-testid="input"
  placeholder="placeholderr"
/>
`;
